general/management/commands/report_count_statuses.py

```diff
@@ -47,13 +49,17 @@ def add_arguments(self, parser):
             action='store_true',
             dest='skip-downloads',
             default=False,
-            help='Using the option --skip-downloads the command will not checked for mismatched downloads (to save time).')
+            help='Using the option --skip-downloads the command will not checked for mismatched downloads '
+                 '(to save time).')
 
     def handle(self,  *args, **options):
 
+        def report_progress(message, total, count):
+            if count % 10000 == 0:
+                console_logger.info(message % (total, 100 * float(count + 1) / total))
+
         # Iterate over all sounds to check: num_comments, num_downloads, avg_rating, num_ratings
         # While iterating, we keep a list of user ids and pack ids for then iterating over them
-        all_user_ids = set()
 
         mismatches_report = {
             'Sound.num_comments': 0,
```

Comment:
mismatches_report and mismatches_object_ids should be converted to a defaultdict to make this MUCH nicer and less maintainance.

```python
from collections import defaultdict
mismatches_report = defaultdict(int)
mismatches_object_ids = defaultdict(list)
```

Reply:
Awesome, fixed it https://github.com/MTG/freesound/pull/1080/commits/e3e54f3b1c4eb3c77c9979266ead374d19a98679
I always forget about these super nice tricks!